### ISSUE TITLE
`tmt tests export --nitrate --create` removes virtual case from fmf file

### ISSUE THREAD

**Issue by lukaszachy** [psss/tmt#238]
At first I had main.fmf like this
```yaml
summary: Some text
contact: my@email.com
component: [ package ]
test: ./test.sh
environment:
    FOO: bar

/additional:
    environment:
        key: value
```
I've run `tmt tests export --nitrate --create` so two cases are created in the nitrate.
But tmt removed virtual case from the fmf file and created just first case (and didn't link it to general plan)
After execution the main.fmf is as following:
```yaml
summary: Some text
contact: my@email.com
component: [ package ]
test: ./test.sh
environment:
    FOO: bar
extra-nitrate: TC#0607295
```

**Comment by psss** [comment]
What you have defined above is just a single test `/additional` which inherits from a non-leaf parent `/`. Handling virtual inheritance using scripts such as `tmt test export` will not be easy but should not be impossible. As a workaround I would suggest to stick with separate file for each virtual test.

**Comment by lukaszachy** [comment]
From my PoV even error message advising me to move `/additional` to a separate file will be enough.
Just not loose my case, please.

Your explanation about single test `/additional` - why is the environment used from the parent then?
IMHO it found 'parent' first and ignored virtual completely.

**Comment by psss** [comment]
@hegerj, @lukaszachy, has this also been covered by #515?

**Comment by hegerj** [comment]
Possibly, but I am not sure. @lukaszachy could you please test it?

**Comment by lukaszachy** [comment]
Well, It does not remove virtual tests any more. However I wouldn't say it works as expected
Tail of main.fmf after the export
```
/additional:
    environment:
        key: value
/bah:
  summary+: other
extra-nitrate: TC#0609454
extra-nitrate: TC#0609455
```
Both virtual cases have same extra-nitrate attribute and it is a duplicate key.

**Comment by psss** [comment]
Thanks for checking this. Seems the only way is to properly modify the virtual hierarchy. Fortunately @jscotka has already prepared a pull request with this functionality in #523. Let's try to go this way.

**Comment by jscotka** [comment]
@lukaszachy Hi Lukas, can you please try my PR's (in FMF, TMT) if it works for you as you've expected

**Comment by lukaszachy** [comment]
So, It should be fixed if one uses tmt with unreleased version of fmf. I'd say that requiring appropriate (>= 0.15) version of fmf is a must to call this fixed properly.

**Comment by psss** [comment]
Ok, agreed. Will be fixed by #674. We need to wait a bit until the new `fmf` reaches stable repos, thus moving to `1.5`.
